ompi/communicator/comm.c

```diff
@@ -1451,7 +1451,7 @@ static int ompi_comm_idup_internal (ompi_communicator_t *comm, ompi_group_t *gro
     }
 
     // Copy info if there is one.
-    {
+    if (info != NULL) {
```

Comment:
I am not sure I understand why this check is required. `ompi_info_memkind_copy_or_set` already has check internally that recognizes if the info object is NULL, see https://github.com/open-mpi/ompi/blob/main/ompi/info/info_memkind.c#L560.

That being said, I see a potential issue a few lines further down in that the argument list of memkind_copy_or_set should take comm->instance->super, instead of comm->super. It looks like I overlooked this location when updating the behaviro to retrieve the value from the instance (instead of the parent communicator)